src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,240 @@
+            {
+                top = mid - 1;
+                foundland = tempPos;
+            }
+            else
+            {
+                bot = mid + 1;
+                foundland = tempPos;
+            }
+            mid = (bot + top)/2;
+        }
+
+        return foundland;
+    }
+
+    /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    private static boolean isPositionSafe(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR
+                && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid()
+                && !sender.getEntityWorld().getBlockState(blockPos.up()).getMaterial().isLiquid();
+    }
+
+    /**
+     * this checks that you are not too close to another colony
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @return colNear false=no true=yes
+     */
+    private static boolean findColony(BlockPos blockPos, World world)
+    {
+        Colony nearestCol = ColonyManager.getClosestColony(world, blockPos);
+        Boolean colNear;
+        /* get individual coords to do the math */
+        int cx = nearestCol != null ? nearestCol.getCenter().getX() : 0;
+        int cz = nearestCol != null ? nearestCol.getCenter().getZ() : 0;
+        /* from the random X for the TP */
+        double px = blockPos.getX();
+        /* from the random Z for the TP */
+        double pz = blockPos.getZ();
+
+        double dist = Math.sqrt(Math.pow(cx - px, 2.0) + (Math.pow(cz - pz, 2.0)));
+                        /* grab the working distance and do our check now that we have distance from nearest colony*/
+                        /* just to understand this better::::
+                            I am taking the working distance from the town hall and doubling it this will give me
+                            the distance needed from both town halls -if you placed it here.  Then im adding on 20%
+                            just to get a padding between the two.*/
+        double wd = (Configurations.workingRangeTownHall * 2) * ADDS_TWENTY_PERCENT;
+        /* bad tp, bad -- abort TP  Too close to a colony */
+        colNear = dist < wd;
+        return colNear;
+    }
+    
+    @NotNull
+    @Override
+    public List<String> getTabCompletionOptions(
+            @NotNull final MinecraftServer server,
+            @NotNull final ICommandSender sender,
+            @NotNull final String[] args,
+            final BlockPos pos)
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public boolean isUsernameIndex(@NotNull String[] args, int index)
+    {
+        return false;
```

Comment:
Yeah that is important command code stuff. You want taht to be yes when the current parameter index could be a player to allow tab autocompletion. In this case always allow it